include/grpc/impl/codegen/port_platform.h

```diff
@@ -337,4 +337,16 @@
   } while (0)
 #endif /* GPR_FORBID_UNREACHABLE_CODE */
 
```

Comment:
I think we want a bit more meat here. Probably something along these lines:

```
#ifdef GPR_WIN32
#  ifdef GRPC_DLL
#    if GRPC_BUILD_LIBRARY
#      define CENSUS_API __declspec(dllexport)
#      define GPR_API __declspec(dllexport)
#      define GRPC_API __declspec(dllexport)
#    else
#      define CENSUS_API __declspec(dllimport)
#      define GPR_API __declspec(dllimport)
#      define GRPC_API __declspec(dllimport)
#    endif
#  else
#    define CENSUS_API
#    define GPR_API
#    define GRPC_API
#  endif
#else /* !GPR_WIN32 */
... what you currently have
#endif
```

Bottom-line, when building grpc from the Makefile or the Visual Studio project for its "dynamic" version, we `-DGRPC_BUILD_LIBRARY=1` and `-DGRPC_DLL=1`, and people who want to use grpc as a DLL compile their code with `-DGRPC_DLL`. This an extremely common way of handling this:

https://github.com/madler/zlib/blob/master/zconf.h#L312